src/material/datepicker/datepicker-intl.ts

```diff
@@ -51,8 +51,19 @@ export class MatDatepickerIntl {
   /** A label for the 'switch to year view' button (used by screen readers). */
   switchToMultiYearViewLabel: string = 'Choose month and year';
 
-  /** Formats a range of years. */
+  /** A label for the first date of a range of dates (used by screen readers). */
+  startDateLabel?: string = 'Start date';
```

Comment:
I don't think that these need to be optional since people are expected to extend the class.

Reply:
The intent is that the labels will not be used if they're not provided. Everything will still work without the labels, but providing them will be better for a11y. That is to make this a non-breaking change. In v15, we would make the labels required by making the property no longer optional.

For example, if we would use startDateLabel as an `aria-description`. if it's undefined, we do do not add the `aria-description` attribute at all.

`[attr.aria-description]: "_intl.startDateLabel"`

Reply:
My point was that it wouldn't be a breaking change even if you made them non-optional and we wouldn't have to null check `formatYearRangeLabel` before calling it.

Reply:
Yeah, I see how it's technically not breaking if you're extending `MatDatepickerIntl` and not implementing. I do like keeping them optional because it gives a way to communicate to developers to implement the new labels. If you extend `MatDatepickerIntl` you will not get any kind of type error when we add new strings.

I think @jelbourn had thoughts on this.

Reply:
I think the making them optional thing was intended as a way to make it non-breaking if we had an abstract class that people extend. Since we've decided not to do that, I'm not sure there's any advantage to making them optional

Reply:
Ohhh, I see why we talked about making them optional. I changed the properties to not optional.